configured_files/config.hpp.in

```diff
@@ -0,0 +1,15 @@
+#ifndef MYPROJECT_CONFIG_HPP
+#define MYPROJECT_CONFIG_HPP
+
+// this is a basic example of how a CMake configured file might look
+// in this particular case, we are using it to set the version number of our executable
+namespace myproject {
+static constexpr std::string_view project_name = "@PROJECT_NAME@";
+static constexpr std::string_view project_version = "@PROJECT_VERSION@";
+static constexpr int project_version_major{@PROJECT_VERSION_MAJOR@};
+static constexpr int project_version_minor{@PROJECT_VERSION_MINOR@};
+static constexpr int project_version_patch{@PROJECT_VERSION_PATCH@};
+static constexpr int project_version_tweak{@PROJECT_VERSION_TWEAK@};
```

Comment:
This is really minor, but it sounds redundant to say `myproject::project_name`, `myproject::project_version`, etc. Maybe it's sufficient to call it `myproject::name`. Calling it `project_name` would be useful if you did something silly like `using namespace myproject;` but nobody's going to do that in cpp-best-practices-town, are they?

Reply:
Hmm... I want to agree with you, but cannot help but feel like `myproject::name` risks collisions, but... this is just a place that the person who uses the template can tweak anyhow.

Reply:
I changed it based on your recommendation

Reply:
Sorry, I'm not trying to make an argument, I'm just trying to understand your point of view. My understanding of the language is not as good as yours; I thought there could be a rule or situation that I'm overlooking. I prefer that you are happy with the code; I may not have considered this thoroughly.

Reply:
Oh, I'm not trying to argue. My thought process was like this:

<user creates project from template>
<user has need for `myprojectnamespace::name`>
<user doesn't know we already provided that, and gets confused>

but I conceded that better naming is probably better

Reply:
```
namespace myproject::cmake {
  static constexpr std::string_view name = "myproject";
}
```

I have no idea at this point. I just know it's handy to have a set of properly typed (type in the C++ sense) set of the common variables from CMake available.

Reply:
Agreed, there's no sense overthinking this. The user will only have trouble with naming collisions if they deliberately `#include <configured_files/config.hpp>`, and they can modify that file if they choose to. If anyone has trouble with a name collision, this should not be too difficult to figure out.

I'm going to call this resolved. I will not object if you want to change the names again.